gradio/queue.py

```diff
@@ -211,70 +240,91 @@ def get_estimation(self) -> Estimation:
+                    )
+                if client_awake:
+                    awake_events.append(event)
+            if not (awake_events):
                 return
             begin_time = time.time()
-            response = await self.call_prediction(event)
+            response = await self.call_prediction(events, batch)
+
             if response.has_exception:
-                await self.send_message(
-                    event,
-                    {
-                        "msg": "process_completed",
-                        "output": {"error": str(response.exception)},
-                        "success": False,
-                    },
-                )
+                for event in awake_events:
+                    await self.send_message(
+                        event,
+                        {
+                            "msg": "process_completed",
+                            "output": {"error": str(response.exception)},
+                            "success": False,
+                        },
+                    )
             elif response.json.get("is_generating", False):
                 while response.json.get("is_generating", False):
                     old_response = response
+                    for event in awake_events:
+                        await self.send_message(
+                            event,
+                            {
+                                "msg": "process_generating",
+                                "output": old_response.json,
+                                "success": old_response.status == 200,
+                            },
+                        )
+                    response = await self.call_prediction(events, batch)
+                for event in awake_events:
                     await self.send_message(
                         event,
                         {
-                            "msg": "process_generating",
+                            "msg": "process_completed",
                             "output": old_response.json,
                             "success": old_response.status == 200,
                         },
                     )
-                    response = await self.call_prediction(event)
-                await self.send_message(
-                    event,
-                    {
-                        "msg": "process_completed",
-                        "output": old_response.json,
-                        "success": old_response.status == 200,
-                    },
-                )
             else:
-                await self.send_message(
-                    event,
-                    {
-                        "msg": "process_completed",
-                        "output": response.json,
-                        "success": response.status == 200,
-                    },
-                )
+                for e, event in enumerate(awake_events):
+                    output = copy.deepcopy(response.json)
```

Comment:
I'm wondering if doing this deep copy for every event will be too wasteful. Imagine if it's a diffusion model with a gallery of 8 images and a batch size of 8. We'd be copying over the 64 images for element in the batch.

Reply:
Yeah let me rewrite this

Reply:
Rewrote to use a single copy